packages/core/echo/echo-db/src/echo-handler/echo-handler.test.ts

```diff
@@ -344,6 +353,33 @@ describe('Reactive Object with ECHO database', () => {
       expect(person.worksAt?.name).to.eq(orgName);
     });
 
+    test('serialized references', async () => {
+      const { db, graph } = await builder.createDatabase();
+      graph.schemaRegistry.addSchema([TestType, TestNestedType]);
+
+      const obj1 = create(TestType, { nested: create(TestNestedType, { field: 'test' }) });
+
+      // Fully serialized before added to db.
+      {
+        const obj = JSON.parse(JSON.stringify(obj1));
+        expect(obj.nested.field).to.eq(obj1.nested?.field);
+      }
+
+      const obj2 = db.add(obj1);
+
+      // References serialized as IPLD.
+      {
+        const obj = JSON.parse(JSON.stringify(obj2));
+        expect(decodeReference(obj.nested).itemId).to.eq(obj2.nested?.id);
+      }
+
+      // Load refs.
+      {
+        const nested = await loadObjectReferences(obj2, (obj) => obj.nested);
```

Comment:
this would have no effect in this test because references are already loaded since you've only just addedthe object to the database

Reply:
we don't have a way to test eviction yet right?